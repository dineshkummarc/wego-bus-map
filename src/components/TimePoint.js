import React from 'react';
import { formatStopTimeUpdate, formatTripTime } from '../util';
import PropTypes from 'prop-types';

function TimePoint({scheduleData, updateData}) {
  // Determine which time point to use for trip update, or none.
  let updateTime = false;
  if (updateData && typeof updateData.departure !== 'undefined' && typeof updateData.departure.time === 'number') {
    updateTime = updateData.departure.time;
  } else if (updateData && typeof updateData.arrival !== 'undefined' && typeof updateData.arrival.time === 'number') {
    updateTime = updateData.arrival.time;
  }

  // Grab the relevant pieces for the scheduled time
  let scheduleTime = formatTripTime(scheduleData.departure_time);

  // Handle case where departure and arrival time are mismatched
  let scheduleDepartNote = '';
  if (scheduleData.arrival_time !== scheduleData.departure_time) {
    scheduleTime = formatTripTime(scheduleData.arrival_time);
    scheduleDepartNote = (<> (Departs {formatTripTime(scheduleData.departure_time)})</>);
  }

  // If no update provided (e.g. past or far-future trip), return scheduled
  if (!updateTime) {
    return(<>{scheduleTime}{scheduleDepartNote}</>);
  }

  updateTime = formatStopTimeUpdate(updateData);

  return(
    <>
      {scheduleTime !== updateTime && (
        <small><strike className="text-muted">{scheduleTime}{scheduleDepartNote}</strike><br /></small>
      )}
      <strong className="text-primary">{updateTime}</strong>
      {scheduleTime === updateTime && (
        <><br /><small>(On Time)</small></>
      )}
    </>
  );
}

TimePoint.propTypes = {
  scheduleData: PropTypes.object.isRequired,
  updateData: PropTypes.object
};

TimePoint.defaultProps = {
  updateData: {}
};

export default TimePoint;
