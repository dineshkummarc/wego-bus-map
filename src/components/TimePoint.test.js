/* globals test */

import React from 'react';
import TimePoint from './TimePoint';
import { createRoot } from 'react-dom/client';

test('renders TimePoint', () => {
  const div = document.createElement('div');
  const root = createRoot(div);
  const scheduleData = {
    "id": 5022349,
    "trip_gid": "265838",
    "trip_id": 153226,
    "arrival_time": " 4:20:00",
    "departure_time": " 4:20:00",
    "stop_gid": "DOVDICWF",
    "stop_id": 46127,
    "stop_sequence": 1,
    "stop_headsign": null,
    "pickup_type": "0",
    "drop_off_type": "0",
    "shape_dist_traveled": null,
    "timepoint": "1",
    "stop": {
      "id": 46127,
      "stop_gid": "DOVDICWF",
      "stop_code": "DOVDICWF",
      "stop_name": "DOVERSIDE DR & DICKERSON PIKE WB",
      "stop_desc": null,
      "stop_lat": "36.24406",
      "stop_lon": "-86.757403",
      "zone_gid": null,
      "stop_url": null,
      "location_type": null,
      "parent_station": null,
      "stop_timezone": null,
      "wheelchair_boarding": "0"
    }
  };
  root.render(<TimePoint scheduleData={scheduleData} />);
});
