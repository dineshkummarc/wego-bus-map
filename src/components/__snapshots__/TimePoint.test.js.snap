// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TimePoint 1`] = `
<div>
  9:55 PM
</div>
`;

exports[`renders TimePoint with trip update 1`] = `
<div>
  <div
    class="small"
  >
    <strike
      class="text-muted"
    >
      9:55 PM
    </strike>
  </div>
  <div>
    <strong
      class="text-primary"
    >
      9:59 PM
    </strong>
  </div>
</div>
`;
